src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let amount = 0;
+
+  const adder = (number) => {
+    if (number === undefined) {
+      const count = amount;
+
+      amount = 0;
+
+      return count;
+    } else {
+      amount += number;
+
+      return adder;
+    }
```

Comment:
You don't need else, bc, if number === undefined - you make return from your function
```suggestion
    }
    
    amount += number;

    return adder;
    }
```